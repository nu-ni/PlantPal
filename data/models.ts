export type PlantCollection = {
    id?: number;
    title: string;
    lastActive: Date;
    count?: number;
};

export type Plant = {
    id?: number;
    title: string;
    frequency: number;
    waterAmount: number;
    collectionId: number;
    image: string;
};

export type Dict = {
    [key: string]: string;
};
