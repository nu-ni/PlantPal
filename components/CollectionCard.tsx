import { Pressable, StyleSheet, Text } from 'react-native';
import { View } from 'react-native';
import React from 'react';
import { PlantCollection } from '../services/Database';

type CardProps = {
  item: PlantCollection;
  isSelected: boolean;
  onItemSelected?: () => void
};

export default function CollectionCard({ item, isSelected, onItemSelected }: CardProps) {
  return (
    <Pressable onPress={onItemSelected}>
      <View style={styles.card}>
        <View style={styles.circleContainer}>
          <View style={[styles.circle, isSelected && styles.circleSelected]}></View>
        </View>
        <View style={styles.divider} />
        <View style={styles.titleContainer}>
          <Text style={styles.title}>{item.title}</Text>
        </View>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  card: {
    backgroundColor: 'white',
    borderRadius: 5,
    flexDirection: 'row',
    padding: 10,
    marginBottom: 10,
  },
  circleContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  circle: {
    width: 20,
    height: 20,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'grey',
  },
  circleSelected: {
    backgroundColor: 'green',
  },
  divider: {
    width: 1,
    height: '100%',
    backgroundColor: 'grey',
  },
  titleContainer: {
    flex: 9,
    justifyContent: 'center',
  },
  title: {
    marginLeft: 10,
  },
});
