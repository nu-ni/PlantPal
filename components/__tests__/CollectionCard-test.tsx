import React from 'react';
import CollectionCard from '../CollectionCard';
import { render, fireEvent } from "@testing-library/react-native";
import renderer from 'react-test-renderer';
import { PlantCollection } from '@/data/models';

// prepare dummy data
const plantCollectionDummy: PlantCollection = {
    id: 1,
    title: "Collection 1",
    lastActive: new Date(),
}

describe("<CollectionCard />", () => {
    // Setup
    const onItemSelectedMock = jest.fn();
    const card = <CollectionCard item={plantCollectionDummy} isSelected={true} onItemSelected={onItemSelectedMock} />;

    // Snapshot test
    it("renders correctly and matches the snapshot", () => {
        const tree = renderer.create(card).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // tests
    it("renders correctly with correct item title displayed", () => {
        const { getByText } = render(card);
        const titleElement = getByText(plantCollectionDummy.title);

        expect(titleElement.props.children).toEqual(plantCollectionDummy.title);
    });

    it("calls onItemSelected when card is pressed", () => {
        const { getByText } = render(card);
        const cardComponent = getByText(plantCollectionDummy.title);

        fireEvent.press(cardComponent);
        expect(onItemSelectedMock).toHaveBeenCalledTimes(1);
    });
});
