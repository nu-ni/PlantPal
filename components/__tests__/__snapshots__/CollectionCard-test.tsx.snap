// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionCard /> renders correctly and matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "elevation": 5,
        "flexDirection": "row",
        "marginBottom": 10,
        "overflow": "visible",
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.45,
        "shadowRadius": 4.84,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginRight": 10,
        }
      }
    >
      <View
        style={
          [
            {
              "borderColor": "grey",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 13,
              "width": 13,
            },
            {
              "backgroundColor": "#557F60",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "#b0b0b0",
          "height": "100%",
          "width": 1,
        }
      }
    />
    <View
      style={
        {
          "flex": 9,
          "justifyContent": "center",
          "marginVertical": 17,
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 13,
            "textAlign": "right",
          }
        }
      >
        Collection 1
      </Text>
    </View>
  </View>
</View>
`;
